src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.sps.data;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.ComputeScopes;
+import com.google.api.services.compute.model.AccessConfig;
+import com.google.api.services.compute.model.CustomerEncryptionKey;
+import com.google.api.services.compute.model.AttachedDisk;
+import com.google.api.services.compute.model.AttachedDiskInitializeParams;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstanceList;
+import com.google.api.services.compute.model.Metadata;
+import com.google.api.services.compute.model.NetworkInterface;
+import com.google.api.services.compute.model.Operation;
+import com.google.api.services.compute.model.ServiceAccount;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.lang.reflect.*;
+
+/** Class that creates, stops and deletes VM instances. */
+public class ComputeAPIClient implements ComputeAPIClientInterface {
+  private static final String APPLICATION_NAME = "VMN";
+  private static final String PROJECT_ID = "cdesir-step-2020";
+  private static final String ZONE_NAME = "us-central1-f";
+  private static final String SOURCE_IMAGE_PREFIX =
+    "https://www.googleapis.com/compute/v1/projects/";
+  private static final String SOURCE_IMAGE_PATH =
+    "cdesir-step-2020/global/images/vmn-image";
+  private static final String NETWORK_INTERFACE_CONFIG = "ONE_TO_ONE_NAT";
+  private static final String NETWORK_ACCESS_CONFIG = "External NAT";
+  private static final long OPERATION_TIMEOUT_MILLIS = 60 * 1000;
+  private static HttpTransport httpTransport;
+  private static final JsonFactory JSON_FACTORY = 
+    JacksonFactory.getDefaultInstance();
+  private static GoogleCredentials credential;
+  private Compute compute;
+  private Operation op;
```

Comment:
Why is this an instance variable? Pass it as a parameter to waitForOperationCompletion instead.